pkg/sql/logictest/testdata/logic_test/fk_opt

```diff
@@ -747,7 +747,7 @@ CREATE TABLE grandchild (id INT PRIMARY KEY, parent_id INT REFERENCES child (par
 statement error pgcode 23503 foreign key
 INSERT INTO grandchild VALUES (1, 1)
 
-statement error "child_parent_id_key" is referenced by foreign key from table "grandchild"
+statement error pq: index "child_parent_id_key" is in use as unique constraint
```

Comment:
you can test for hints if you want: https://github.com/cockroachdb/cockroach/blob/fe74126684d1bcc39d7c8477a83e5b6e58afe957/pkg/sql/logictest/testdata/logic_test/array#L418